mods/parsoid.js

```diff
@@ -444,6 +444,16 @@ PSP.transformRevision = function (restbase, req, from, to) {
 
     return this._getOriginalContent(restbase, req, rp.revision, tid)
     .then(function (original) {
+        // Check if parsoid metadata is present as it's required by parsoid.
+        if (!original || !original['data-parsoid'] || !original['data-parsoid'].body.ids) {
```

Comment:
`original` and `original['data-parsoid']` do not need to be checked for existence. However, before checking for `original['data-parsoid'].body.ids` you should check for `original['data-parsoid'].body` since that is the part that is `undefined` if *data-parsoid* is missing from storage.